internal/downloader/download.go

```diff
@@ -111,14 +139,19 @@ func (d *Downloder) download(desc, urlToGet, destination string, mode os.FileMod
 			resp.Status,
 		)
 	}
+	temporaryDestinationFile, err := ioutil.TempFile(os.TempDir(), "kuberlr-kubectl-");
+	if err != nil {
+		return fmt.Errorf("Error trying to create temporary file in %s: %v", os.TempDir(), err)
+	}
 
-	f, err := os.OpenFile(destination, os.O_CREATE|os.O_WRONLY, mode)
+	f, err := os.OpenFile(temporaryDestinationFile.Name(), os.O_CREATE|os.O_WRONLY, mode)
 	if err != nil {
 		return fmt.Errorf(
 			"Error while downloading %s to %s: %v",
-			urlToGet, destination, err)
+			urlToGet, temporaryDestinationFile.Name(), err)
 	}
-	defer f.Close()
+	defer temporaryDestinationFile.Close()
+	defer os.Remove(temporaryDestinationFile.Name())
```

Comment:
I would not remove these lines, see below for an explanation.

Reply:
I don't see the problem with the defers -- I've verified that the temp file is always deleted.

Reply:
You're right, `os.Remove` will probably return an `err` when the file is moved, but we don't check this return value. Hence it's fine to leave the defers around